applications/rag/example_notebooks/rag-ray-ingest-with-kubernetes-docs.ipynb

```diff
@@ -0,0 +1,291 @@
+    "VECTOR_EMBEDDINGS_TABLE_NAME = \"rag_embeddings_db\"\n",
+    "DB_NAME = \"pgvector-database\"\n",
+    "\n",
+    "db_username_file = open(\"/etc/secret-volume/username\", \"r\")\n",
+    "DB_USER = db_username_file.read()\n",
+    "db_username_file.close()\n",
+    "\n",
+    "db_password_file = open(\"/etc/secret-volume/password\", \"r\")\n",
+    "DB_PASS = db_password_file.read()\n",
+    "db_password_file.close()\n",
+    "\n",
+    "# initialize Connector object\n",
+    "connector = Connector()\n",
+    "\n",
+    "# function to return the database connection object\n",
+    "def getconn():\n",
+    "    conn = connector.connect(\n",
+    "        INSTANCE_CONNECTION_NAME,\n",
+    "        \"pg8000\",\n",
+    "        user=DB_USER,\n",
+    "        password=DB_PASS,\n",
+    "        db=DB_NAME,\n",
+    "        ip_type=IPTypes.PRIVATE\n",
+    "    )\n",
+    "    return conn\n",
+    "\n",
+    "# create connection pool with 'creator' argument to our connection object function\n",
+    "pool = sqlalchemy.create_engine(\n",
+    "    \"postgresql+pg8000://\",\n",
+    "    creator=getconn,\n",
+    ")\n",
+    "\n",
+    "Base = declarative_base()\n",
+    "DBSession = scoped_session(sessionmaker())\n",
+    "\n",
+    "class TextEmbedding(Base):\n",
+    "    __tablename__ = VECTOR_EMBEDDINGS_TABLE_NAME\n",
+    "    id = Column(String(255), primary_key=True)\n",
+    "    text = Column(Text)\n",
+    "    text_embedding = mapped_column(Vector(384))\n",
+    "\n",
+    "with pool.connect() as conn:\n",
+    "    conn.execute(text(\"CREATE EXTENSION IF NOT EXISTS vector\"))\n",
+    "    conn.commit() \n",
+    "    \n",
+    "DBSession.configure(bind=pool, autoflush=False, expire_on_commit=False)\n",
+    "Base.metadata.drop_all(pool)\n",
+    "Base.metadata.create_all(pool)\n",
+    "\n",
+    "SENTENCE_TRANSFORMER_MODEL = \"intfloat/multilingual-e5-small\"  # Transformer to use for converting text chunks to vector embeddings\n",
+    "\n",
+    "# the dataset has been pre-dowloaded to the GCS bucket as part of the notebook in the cell above. Ray workers will find the dataset readily mounted.\n",
+    "SHARED_DATASET_BASE_PATH = \"/data/kubernetes-docs/\"\n",
+    "\n",
+    "CHUNK_SIZE = 1000  # text chunk sizes which will be converted to vector embeddings\n",
+    "CHUNK_OVERLAP = 10\n",
+    "VECTOR_DIMENSION = 384  # Embeddings size\n",
+    "\n",
+    "splitter = RecursiveCharacterTextSplitter(chunk_size=CHUNK_SIZE, chunk_overlap=CHUNK_OVERLAP, length_function=len)\n",
+    "embeddings_service = HuggingFaceEmbeddings(model_name=SENTENCE_TRANSFORMER_MODEL)\n",
+    "\n",
+    "def process_pdf(file_path):\n",
+    "    \"\"\"Loads, splits and embed a single PDF file.\"\"\"\n",
+    "    loader = PyPDFLoader(file_path)\n",
+    "    print(f\"Loading {file_path}\")\n",
+    "    pages = loader.load_and_split()\n",
+    "    \n",
+    "    splits = splitter.split_documents(pages)\n",
+    "\n",
+    "    chunks = []\n",
```

Comment:
The variable name `chunks` is confusing here, it sounds more like the raw data chunks before embedding